services/cache/lru.go

```diff
@@ -221,6 +221,11 @@ func (l *LRU) get(key string, value interface{}) error {
 			_, err := s.UnmarshalMsg(e.value)
 			*v = &s
 			return err
+		case *map[string]*model.User:
+			var u model.UserMap
+			_, err := u.UnmarshalMsg(e.value)
```

Comment:
Unrelated question but since we are here I'd rather ask:
I understand a lot of locking pressure will be eased with the upcoming striped LRU implementation but did we ever consider doing the serialization/deserialization operations out of the critical section?

Reply:
Nice one! Can you please file a ticket. I will take care of it.

Reply:
https://mattermost.atlassian.net/browse/MM-30140